charms/istio-pilot/src/charm.py

```diff
@@ -836,21 +810,21 @@ def _ssl_info(self) -> Dict[str, str]:
         # FIXME: remove the try/catch block and just return the dictionary that contains
         # data from the CertHandler after 1.21
         try:
-            ssl_secret = self.model.get_secret(label=TLS_SECRET_LABEL)
+            tls_secret = self.model.get_secret(id=self._tls_secret_id, label=TLS_SECRET_LABEL)
```

Comment:
I'm asking Juju as well, but do you see a reason why the `label` helps us?  afaict, we never know if this is the first time we're accessing the secret, so we never know if we need to set the label.  To me I think we should just remove the label entirely since we never access the secret by label.  wdyt?

Reply:
That is true, I was also debating to keep this or not. The only reason to keep it is to prevent users from passing the ID of another secret.

Reply:
I have removed the label entirely, nothing else to do. Resolving this conversation.